### ISSUE TITLE
On touchscreens the dragged element gets temporarily frozen when the element 'touchstart' was fired on is removed from the dom

### ISSUE THREAD

**Issue by isaacHagoel** [isaacHagoel/svelte-dnd-action#58]
it is possible to pick it back up for some reason.
maybe it is because the elements are too small

**Comment by zba** [comment]
It always happen when element leaves source dragzone, after start to drag it again, it never froze,

**Comment by zba** [comment]
It happens when dragged node removed, so touch start to be "document's" action, if you have a scroll in document, you will easy find that it start to scroll document instead of continue touch.

**Comment by isaacHagoel** [comment]
@zba great insight! So this happens when Svelte removes the shadow item from the origin list.
I have confirmed that:
1. if i don't remove the shadow element from the origin dnd-zone, the bug doesn't happen (it causes other problems though but nothing unfixable)
2. if i add a delay on the consider event when the dragged element leaves the origin dnd-zone, it freezes after the delay (so this is not a potential fix but further proves the point)

I tried moving the touchstart eventlistener to the window level (it was checking whether it should trigger a drag based on e.target.patentElement). It was working like the current version but didn't fix the problem.
I will keep looking for a feasible solution, open to ideas...

**Comment by isaacHagoel** [comment]
I've spent two hours or so trying to find the root cause and doing all kind of crazy experiments. no luck so far.

**Comment by zba** [comment]
I not very great in svelte, but ppl on internet mean something about index, so may be it possible to render shadow inside single {each ()} block ?

**Comment by isaacHagoel** [comment]
@zba okay. Good news. After hours of experimenting and googling I have found why this happens.
I have reproduced it here:
https://svelte.dev/repl/974ded85c200451ab7b3eb14b1a6b1d7?version=3.23.0
It has nothing to do with Svelte. It is a touch events "feature".
As insane as it may be, it turns out that touch events are unlike mouse events in that they always use the element on which the touch started as their target, even after the touch moved away, even though the touch listeners are not attached to that original element at all (in my example I attach to the parent and it makes no difference).
This means that if that original element is ever removed while touch is ongoing, it will interrupt the flow of touch events, which is what we see.

The exact same issue existed for this library: 
https://github.com/timruffles/mobile-drag-drop/issues/143

and the maintainer provides two great links. Mainly [this one](https://docs.google.com/document/d/12-HPlSIF7-ISY8TQHtuQ3IqDi-isZVI0Yzv5zwl90VU/edit#heading=h.q2zqz8v0mja7).

This ugly piece of code makes the problem go away:
```javascript
        // creating the draggable element
        draggedEl = createDraggedElementFrom(dragTarget);
        // this is a workaround 
       window.setTimeout(() => {
           document.body.appendChild(draggedEl);
            dragTarget.style.display = 'none';
            dragTarget.style.position = 'fixed';
            dragTarget.style.zIndex = '-5';
            document.body.appendChild(dragTarget);
        }, 0);
```
Notice that i also have to append the clone inside of the timeout callback in order for this to work. I am not sure why that is.
I will find some time over the coming days to turn this into a proper fix (need to put this functionality in `styler.js`, clean up this silly element on drop, test more).
**At least now there is a clear path to resolving this annoying behaviour** 🤖
